packages/aws-cdk-lib/aws-eks/README.md

```diff
@@ -1100,6 +1103,104 @@ consoleReadOnlyRole.addToPolicy(new iam.PolicyStatement({
 cluster.awsAuth.addMastersRole(consoleReadOnlyRole)
 ```
 
+### Access Config
+
+Amazon EKS supports three modes of authentication: `CONFIG_MAP`, `API_AND_CONFIG_MAP`, and `API`. You can enable cluster
+to use access entry APIs by using authenticationMode `API` or `API_AND_CONFIG_MAP`. Use authenticationMode `CONFIG_MAP`
+to continue using aws-auth configMap exclusively. When `API_AND_CONFIG_MAP` is enabled, the cluster will source authenticated
+AWS IAM principals from both Amazon EKS access entry APIs and the aws-auth configMap, with priority given to the access entry API.
+
+To specify the `authenticationMode`:
+
+```ts
+import { KubectlV29Layer } from '@aws-cdk/lambda-layer-kubectl-v29';
+declare const vpc: ec2.Vpc;
+
+new eks.Cluster(this, 'Cluster', {
+  vpc,
+  version: eks.KubernetesVersion.V1_29,
+  kubectlLayer: new KubectlV29Layer(this, 'KubectlLayer'),
+  authenticationMode: eks.AuthenticationMode.API_AND_CONFIG_MAP,
+});
+```
+
+> **Note** - Switching authentication modes on an existing cluster is a one-way operation. You can switch from
+> `CONFIG_MAP` to `API_AND_CONFIG_MAP`. You can then switch from `API_AND_CONFIG_MAP` to `API`.
+> You cannot revert these operations in the opposite direction. Meaning you cannot switch back to
```

Comment:
For my learning purposes, what if I switch it in opposite direction? Is it going to cause a deployment failure or a resoruce replacement?

Reply:
@GavinZZ Thank you for bringing it up.

Amazon EKS does not allow you to turn off the `API` support after you opt in or re-enable the ConfigMap after you disabled it. That is,

❌  `API_AND_CONFIG_MAP` to `CONFIG_MAP` -> not allowed as you are turning off the API support
❌  `API` to `CONFIG_MAP` -> not allowed as you are turning off the API support and also you are re-enabling the config map from disabled.
❌  `API` to `API_AND_CONFIG_MAP` -> not allowed as you are enabling the ConfigMap from disabled.
❌  `API_AND_CONFIG_MAP` to `CONFIG_MAP` or `undefined` -> not allowed as you are disabling API from enabled.
❌  Any defined value to undefined -> not allowed, EKS API does not allow you to switch back once you opt in.

This PR essentially blocks all those cases in the CDK code but we can only detect that in the custom resource so users still have to deploy it through the cloudformation and the custom resource would throw. 

When this happen:

1. no resource would be replaced
2. no SDK calls would happen because CDK custom resource would throw first